x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_editing/index.tsx

```diff
@@ -398,43 +440,39 @@ const EditRulePageComponent: FC<{ rule: RuleResponse }> = ({ rule }) => {
       scheduleStepFormValid &&
       actionsStepFormValid
     ) {
-      startTransaction({ name: SINGLE_RULE_ACTIONS.SAVE });
-      await updateRule({
-        ...formatRule<RuleUpdateProps>(
-          defineStepData,
-          aboutStepData,
-          scheduleStepData,
-          actionsStepData,
-          triggersActionsUi.actionTypeRegistry,
-          rule?.exceptions_list
-        ),
-        ...(ruleId ? { id: ruleId } : {}),
-      });
+      await saveChanges();
+      return;
+    }
 
-      displaySuccessToast(i18n.SUCCESSFULLY_SAVED_RULE(rule?.name ?? ''), dispatchToaster);
-      navigateToApp(APP_UI_ID, {
-        deepLinkId: SecurityPageName.rules,
-        path: getRuleDetailsUrl(ruleId ?? ''),
-      });
+    const { blockingErrors, nonBlockingErrors } = getRuleFormsErrors({
+      defineStepForm,
+      aboutStepForm,
+      scheduleStepForm,
+      actionsStepForm,
+    });
+    if (blockingErrors.length > 0) {
+      return;
+    }
+    if (nonBlockingErrors.length > 0) {
+      setNonBlockingRuleErrors(nonBlockingErrors);
+      showSaveWithErrorsModal();
     }
   }, [
     defineStepForm,
     aboutStepForm,
     scheduleStepForm,
     actionsStepForm,
-    startTransaction,
-    updateRule,
-    defineStepData,
-    aboutStepData,
-    scheduleStepData,
-    actionsStepData,
-    rule,
-    ruleId,
-    dispatchToaster,
-    navigateToApp,
-    triggersActionsUi.actionTypeRegistry,
+    getRuleFormsErrors,
+    saveChanges,
   ]);
 
+  const showSaveWithErrorsModal = () => setIsSaveWithErrorsModalVisible(true);
+  const closeSaveWithErrorsModal = () => setIsSaveWithErrorsModalVisible(false);
```

Comment:
Component `SaveWithErrorsModal` (x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/save_with_errors_confirmation/index.tsx) is wrapped in memo, but these callbacks will be recreated on every re-render. So, memo won't work as intended

Reply:
same for `x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/index.tsx` file as well

Reply:
good catch!